taichi/transforms/alg_simp.cpp

```diff
@@ -341,7 +358,15 @@ bool use_fast_math(IRNode *root) {
 
 bool alg_simp(IRNode *root) {
   TI_AUTO_PROF;
-  return AlgSimp::run(root, hack::use_fast_math(root));
+  PassContext ctx;
+  if (root->get_kernel())
+    ctx.config_ = root->get_config();
+  else
+    ctx.config_.fast_math = false;
+  PassManager mgr;
+  AlgSimpPass::run(ctx, root, &mgr);
+  return mgr.get_pass_result<AlgSimpPass>().modified;
```

Comment:
I will rewrite this when we deprecate the API `bool alg_simp(IRNode *root)`. Will `TI_AUTO_PROF` still look good if we move it to `static void AlgSimpPass::run`?

Reply:
`TI_AUTO_PROF` uses `__FUNCTION__` as the profiled function name, so maybe we'll have a lot of `run()` in the profiler's output...

Reply:
Then just use `TI_PROFILE` with a more explicit name?

Reply:
Yeah, this is doable.